dashboard/app/models/user.rb

```diff
@@ -245,6 +248,15 @@ def save_parent_email_preference
     end
   end
 
+  # Enables/disables sharing of emails of teachers in the U.S. to Code.org regional partners based on user's choice.
+  def save_email_reg_partner_preference
```

Comment:
I _believe_ since this will be ran `after_save`, every update to the user will trigger this method. Therefore the date time would be overwritten here on each update. Might be worth testing that locally, and I noticed the other preferences use [this method that is checking if a value exists already for reference.](https://github.com/code-dot-org/code-dot-org/blob/share-teacher-email-regional-partner-opt/dashboard/app/models/email_preference.rb#L41)

Reply:
Using `after_create` instead might be the best solution. Do we plan to allow the user to change this preference if they want?

Reply:
The `after_save` is only triggered when the `:save_email_reg_partner_preference` is set upon a user being created (so it will only be triggered once when the user finishes signing up). I double checked with testing that it only updates the once!

And currently the plan for this PR is just to set up the opt-in preference on the sign-up page. In the future there may be a separate objective to allow a user to modify this choice but for now Bryan just wants to see how this change alone will affect user sign-up and user experience before putting more time into it.

Reply:
Ahh okay nice! 👍 

It might make sense to add the `else`(not a teacher or `share_teacher_email_reg_partner_opt_in_radio_choice` is "no") scenario and set `share_teacher_email_regional_partner_opt_in` to `nil`. As long as their are no complications that you can think of - it's a quick addition and I believe would future proof this/allow a user to be set from opted in to not opted in.